hapi-fhir-base/src/main/java/ca/uhn/fhir/parser/BaseParser.java

```diff
@@ -183,6 +183,19 @@ protected Iterable<CompositeChildElement> compositeChildIterator(
 				});
 	}
 
+	/**
+	 * We add the reference to the input resources to ensure it doesn't get
+	 * overwritten if it wasn't there.
+	 * The resource is being mutated anyways, so this is fine.
+	 *
+	 * We need to maintain the reference as an internal reference (ie,
+	 * preceded by an #) because otherwise later transactions cannot
+	 * process the resource (since the reference will not be set).
+	 */
+	private void setReference(IBaseReference theReference, String theText) {
```

Comment:
suggestion, non-blocking: `theText` is not that descriptive
```suggestion
	private void setReference(IBaseReference theReference, String theReferenceId) {
```